### ISSUE TITLE
Generated NavType for Serializable types do not compile for version 1.4.0+

### ISSUE THREAD

**Issue by dpredovic** [raamcosta/compose-destinations#84]
1.4.0+  generated NavType:

```
class LocalTimeNavType(
    private val stringSerializer: DestinationsNavTypeSerializer<Serializable>
) : DestinationsNavType<LocalTime?>() {

    override fun get(bundle: Bundle, key: String): LocalTime? {
        return bundle.getSerializable(key) as LocalTime?
    }

    override fun put(bundle: Bundle, key: String, value: LocalTime?) {
        bundle.putSerializable(key, value)
    }

    override fun parseValue(value: String): LocalTime {
        return stringSerializer.fromRouteString(value)
    }

    override fun serializeValue(value: LocalTime?): String? =
        value?.let { encodeForRoute(stringSerializer.toRouteString(value)) }
}

```
1.3.5 generated NavType:
```
class LocalTimeNavType(
    private val stringSerializer: SerializableNavTypeSerializer<Serializable>
) : SerializableDestinationsNavType<LocalTime?>() {

    override fun get(bundle: Bundle, key: String): LocalTime? {
        return bundle.getSerializable(key) as LocalTime?
    }

    override fun put(bundle: Bundle, key: String, value: LocalTime?) {
        bundle.putSerializable(key, value)
    }

    override fun parseValue(value: String): LocalTime {
        return stringSerializer.fromRouteString(value) as LocalTime
    }

    override fun serializeValue(value: Serializable): String {
        return encodeForRoute(stringSerializer.toRouteString(value))
    }
}

```
1.4.0+ generated parseValue misses casting?

**Comment by raamcosta** [comment]
Daaamn 🤦‍♂️

This was a late change we had from one of our contributors, but I definitely should have tested.
I’ll make this specific case part of the sample app so that this doesn’t happen again and I’ll make a new release with the fix today still.

Thank you for reporting! 🙇‍♂️

**Comment by raamcosta** [comment]
Fixed, it will be part of 1.4.2-beta coming later today. Thanks again @dpredovic! 🙇
